x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/WaterfallContainer/Waterfall/WaterfallItem.tsx

```diff
@@ -154,12 +167,20 @@ export function WaterfallItem({
   const width = (item.duration / totalDuration) * 100;
   const left = ((item.offset + item.skew) / totalDuration) * 100;
 
+  const [isHovering, setIsHovering] = useState(false);
+
   return (
     <Container
       type={item.docType}
       timelineMargins={timelineMargins}
       isSelected={isSelected}
       onClick={onClick}
+      onMouseEnter={() => {
+        setIsHovering(true);
+      }}
+      onMouseLeave={() => {
+        setIsHovering(false);
+      }}
```

Comment:
Doing this with js could be a little sluggish, since the entire component is re-rendered on every hover. What about some good ol' css?

```less
.parent-container {
  .child-container {
    display: none;
  }

  &:hover .child-container {
    display: block;
  }
}
```

Reply:
Btw. I noticed that it feels pretty snappy - so if the css approach it not doable it's probably fin as-is.